src/main/java/com/softserve/academy/spaced/repetition/domain/RememberingLevel.java

```diff
@@ -50,4 +50,19 @@ public Integer getNumberOfPostponedDays() {
     public void setNumberOfPostponedDays(Integer numberOfPostponedDays) {
         this.numberOfPostponedDays = numberOfPostponedDays;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        RememberingLevel that = (RememberingLevel) o;
+
+        return id.equals(that.id);
```

Comment:
A NullPointerException is possible here, isn't it?

Reply:
Yes, of course. And what do you mean?

Reply:
Please rewrite the code in a way that no NPE will be thrown.

Reply:
Sorry but I think we need it in order to have it in our exception handler controller so that the right response http code could be returned. Otherwise, I could return hashcode of id only when id != 0 but what am I supposed to do in the other case?

Reply:
Read any "equals and hashcode best practice".

Or just replace the line by
`return id == null ? false : id.equals(that.id);`

Or (for Java 8 and above) by
`return Objects.equals(id, that.id);`

There are plenty of options how to avoid NPE both inside equals() and inside hashcode().

Reply:
Well, I don't think it will be correct. There can be situations when the object is not saved yet, therefore it doesn't have an id but logically it's equal to another object.
There are plenty of practices how to fix that but there no the only one that is the best and suitable to all the cases.
I could rewrite equals and hashcode methods due to hibernate advices so that it'd compare objects using so called business keys instead of ids. In my case business keys for the UserCardQueue objects will be accountEmail and cardId, which, we suppose, should never be null or NullPointerException must be thrown.
So, to make the long story short, I'm not sure which practice to use in that case and I'm asking you fot your piece of advice. Moreover, you said it was better to compare objects using their ids, so I'd like not to override that without your permission.

Reply:
I'm keeping my different opinion. If the object is not saved yet it is not equal to any other (except himself) and no problems.